core-experimental/src/RotationSemantics.cpp

```diff
@@ -54,47 +61,40 @@ namespace iDynTree
 
     void RotationSemantics::setReferenceOrientationFrame(int _refOrientationFrame)
     {
-        this->refOrientationFrame = _refOrientationFrame;
+        this->refOrientationFrame = this->coordinateFrame = _refOrientationFrame;
```

Comment:
Good way of enforcing the constraint on semantic, but we should mention it in the doxygen documentation, otherways this behaviour is really counterintuitive.